test/linalg/dense.jl

```diff
@@ -185,354 +192,358 @@ for elty in (Int32, Int64, Float32, Float64, Complex64, Complex128)
+            @test norm(As, 1) ≈ 5
+            elty <: Union{BigFloat,Complex{BigFloat},BigInt} || @test norm(As, 2) ≈ 5
+            @test norm(As, Inf) ≈ 5
+        end
 
-            # vecnorm:
-            for p = -2:3
-                @test norm(reshape(A, length(A)), p) == vecnorm(A, p)
+        @testset "Absolute homogeneity, triangle inequality, & vecnorm" begin
+            for i = 1:10
+                Ainit = elty <: Integer ? convert(Matrix{elty}, rand(1:10, mmat, nmat)) :
+                        elty <: Complex ? convert(Matrix{elty}, complex.(randn(mmat, nmat), randn(mmat, nmat))) :
+                        convert(Matrix{elty}, randn(mmat, nmat))
+                Binit = elty <: Integer ? convert(Matrix{elty}, rand(1:10, mmat, nmat)) :
+                        elty <: Complex ? convert(Matrix{elty}, complex.(randn(mmat, nmat), randn(mmat, nmat))) :
+                        convert(Matrix{elty}, randn(mmat, nmat))
+                α = elty <: Integer ? randn() :
+                    elty <: Complex ? convert(elty, complex(randn(),randn())) :
+                    convert(elty, randn())
+
+                for arraytype in ("Array", "SubArray")
+                    if arraytype == "Array"
+                        A = Ainit
+                        B = Binit
+                    else
+                        A = view(Ainit,1:nmat,1:nmat)
+                        B = view(Binit,1:nmat,1:nmat)
+                    end
+
+                    # Absolute homogeneity
+                    @test norm(α*A,1) ≈ abs(α)*norm(A,1)
+                    elty <: Union{BigFloat,Complex{BigFloat},BigInt} || @test norm(α*A) ≈ abs(α)*norm(A) # two is default
+                    @test norm(α*A,Inf) ≈ abs(α)*norm(A,Inf)
+
+                    # Triangle inequality
+                    @test norm(A + B,1) <= norm(A,1) + norm(B,1)
+                    elty <: Union{BigFloat,Complex{BigFloat},BigInt} || @test norm(A + B) <= norm(A) + norm(B) # two is default
+                    @test norm(A + B,Inf) <= norm(A,Inf) + norm(B,Inf)
+
+                    # vecnorm:
+                    for p = -2:3
+                        @test norm(reshape(A, length(A)), p) == vecnorm(A, p)
+                    end
+                end
             end
-        end
 
-        # issue #10234
-        if elty <: AbstractFloat || elty <: Complex
-            let z = zeros(elty, 100)
-                z[1] = -Inf
-                for p in [-2,-1.5,-1,-0.5,0.5,1,1.5,2,Inf]
-                    @test norm(z, p) == (p < 0 ? 0 : Inf)
-                    @test norm(elty[Inf],p) == Inf
+            @testset "issue #10234" begin
+                if elty <: AbstractFloat || elty <: Complex
+                    z = zeros(elty, 100)
+                    z[1] = -Inf
+                    for p in [-2,-1.5,-1,-0.5,0.5,1,1.5,2,Inf]
+                        @test norm(z, p) == (p < 0 ? 0 : Inf)
+                        @test norm(elty[Inf],p) == Inf
+                    end
```

Comment:
Changes to this section of code (lines 335 to 342) are causing the reduction in the number of tests. This code block was formerly wrapped into the `for` loop starting on line 298 (taking `i` from 1 to 10), and I removed it when I added testsets. 

It looks to me like the same tests at lines 340 and 341 were being repeated 10 times (for each type) as they don't appear to have any dependency on `i`, `A`, `B`, or `α`. Does it look like I understand correctly? Thank you!